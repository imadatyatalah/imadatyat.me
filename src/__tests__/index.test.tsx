import React from "react";

import { render, screen } from "@testing-library/react";

import Home from "@/pages/index";

describe("Home", () => {
  it("renders a heading", () => {
    render(<Home />);

    const heading = screen.getByRole("heading", {
      name: "Next Themes + Tailwind Dark Mode",
    });

    expect(heading).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    render(<Home />);

    expect(screen.getByTestId("home-page")).toMatchSnapshot();
  });
});
