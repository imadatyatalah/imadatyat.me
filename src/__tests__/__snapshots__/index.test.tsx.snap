// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<section
  data-testid="home-page"
>
  <div
    class="flex flex-col items-center justify-center py-20"
  >
    <h1
      class="text-5xl font-bold text-center text-gray-800 dark:text-gray-100"
    >
      Next Themes + Tailwind Dark Mode
    </h1>
    <button
      class="px-4 py-2 mt-16 font-semibold text-white bg-black rounded-md dark:text-black dark:bg-white"
    >
      Change Theme
    </button>
  </div>
</section>
`;
